src/cider/nrepl/middleware/stacktrace.clj

```diff
@@ -68,10 +74,50 @@
   (let [tool-regex #"^clojure\.lang\.Compiler|^clojure\.tools\.nrepl|^cider\."
         tool? #(re-find tool-regex (or (:name %) ""))
         flag  #(if (tool? %)
-                 (update-in % [:flags] (comp set conj) :tooling)
+                 (flag-frame % :tooling)
                  %)]
     (map flag frames)))
 
+(def directory-namespaces
+  "This looks for all namespaces inside of directories on the class
+  path, ignoring jars."
+  (let [paths (-> (System/getProperty "java.class.path")
+                  (str/split #":"))]
+    (->> paths
+         (filter (complement #(.endsWith % ".jar")))
+         (map io/file)
+         (filter #(.isDirectory %))
+         (nsfind/find-namespaces)
+         (into #{}))))
+
+(def ns-common-prefix
+  "In order to match more namespaces, we look for a common namespace
+  prefix across the ones we have identified."
+  (let [common
+        (try (reduce
+              (fn [common ns]
+                (let [ns (str/lower-case ns)
+                      matched (map vector common ns)
+                      coincident (take-while (fn [[a b]] (= a b)) matched)]
+                  (apply str (map first coincident))))
+              (str/lower-case (first directory-namespaces))
+              directory-namespaces)
+             (catch Throwable e "NO_PREFIX_INFO"))]
```

Comment:
that's true, but that would add logic into whether we _should_ try to match the namespace instead of letting it just fail to match. I'm happy to change it, I'm just wondering what it would look like. Perhaps let it coalesce to nil and just add a `when` clause? I just didn't want to accidentally successfully match the empty string against every namespace and mark everything as the project, which I thought this method handled explicitly